src/state/slices/common-selectors.ts

```diff
@@ -196,6 +196,17 @@ export const selectAssetsSortedByName = createDeepEqualOutputSelector(selectAsse
   return orderBy(Object.values(assets).filter(isSome), [getAssetName], ['asc'])
 })
 
+export const selectAssetsSortedByNameAndBalance = createDeepEqualOutputSelector(
+  selectAssetsSortedByName,
+  selectPortfolioUserCurrencyBalances,
+  (assets, portfolioUserCurrencyBalances) => {
+    const getAssetBalance = (asset: Asset) =>
+      bnOrZero(portfolioUserCurrencyBalances[asset.assetId]).toNumber()
+
+    return orderBy(Object.values(assets).filter(isSome), [getAssetBalance], ['desc'])
+  },
+)
+
```

Comment:
non-blocking, suggestion: this will iterate twice and assets list can be quite big - I would actually go against the DRY conventions here and do the logic in-place in this selector

```suggestion
export const selectAssetsSortedByNameAndBalance = createDeepEqualOutputSelector(
  selectAssets,
  selectPortfolioUserCurrencyBalances,
  (assets, portfolioUserCurrencyBalances) => {
    const getAssetName = (asset: Asset) => asset.name.toLowerCase()
    const getAssetBalance = (asset: Asset) =>
      bnOrZero(portfolioUserCurrencyBalances[asset.assetId]).toNumber()
    
    return orderBy(
      Object.values(assets).filter(isSome),
      [getAssetBalance, getAssetName],
      ['desc', 'asc']
    )
  }
)
```

Reply:
Done in [e43ba10](https://github.com/shapeshift/web/pull/7546/commits/e43ba107bbc592794b2fa749985f6bc84b1fd463)

Here is the final result:
<img width="493" alt="image" src="https://github.com/user-attachments/assets/d42deb10-b8e3-4f55-9518-e8c74bc9ffdb">